packages/client/src/client.ts

```diff
@@ -72,6 +88,11 @@ export class DefaultDelegatedRoutingV1HttpApiClient implements DelegatedRoutingV
     this.httpQueue.clear()
     this.shutDownController.abort()
     this.started = false
+
+    // Clear the cache when stopping
+    if (this.cache != null) {
+      void this.cache.delete('delegated-routing-v1-cache')
```

Comment:
I just realised that this can be a problem if you have multiple instances of the client in scope as we do in Verified Fetch. 

But even then, it's most likely that they all get stopped at the same time.